hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/SingleTransactionRecordBuilderImpl.java

```diff
@@ -91,412 +89,722 @@ public class SingleTransactionRecordBuilderImpl
+    private List<AccountAmount> paidStakingRewards = new ArrayList<>();
+    private final TransactionRecord.Builder transactionRecordBuilder = TransactionRecord.newBuilder();
+
     // fields needed for TransactionReceipt
     private ResponseCodeEnum status = ResponseCodeEnum.OK;
-    private AccountID accountID;
-    private FileID fileID;
-    private ContractID contractID;
-    private ExchangeRateSet exchangeRate;
-    private TopicID topicID;
-    private long topicSequenceNumber;
-    private Bytes topicRunningHash = Bytes.EMPTY;
-    private long topicRunningHashVersion;
-    private TokenID tokenID;
-    private long newTotalSupply;
-    private ScheduleID scheduleID;
-    private TransactionID scheduledTransactionID;
-    private List<Long> serialNumbers = emptyList();
+    private List<Long> serialNumbers = new ArrayList<>();
+    private final TransactionReceipt.Builder transactionReceiptBuilder = TransactionReceipt.newBuilder();
     // Sidecar data, booleans are the migration flag
-    public final List<AbstractMap.SimpleEntry<ContractStateChanges, Boolean>> contractStateChanges = new ArrayList<>();
-    public final List<AbstractMap.SimpleEntry<ContractActions, Boolean>> contractActions = new ArrayList<>();
-    public final List<AbstractMap.SimpleEntry<ContractBytecode, Boolean>> contractBytecodes = new ArrayList<>();
+    private List<AbstractMap.SimpleEntry<ContractStateChanges, Boolean>> contractStateChanges = new ArrayList<>();
+    private List<AbstractMap.SimpleEntry<ContractActions, Boolean>> contractActions = new ArrayList<>();
+    private List<AbstractMap.SimpleEntry<ContractBytecode, Boolean>> contractBytecodes = new ArrayList<>();
 
+    /**
+     * Creates new transaction record builder.
+     *
+     * @param consensusNow the consensus timestamp for the transaction
+     */
     public SingleTransactionRecordBuilderImpl(@NonNull final Instant consensusNow) {
         this.consensusNow = requireNonNull(consensusNow, "consensusNow must not be null");
-        this.parentConsensusTimestamp = null;
+        this.parentConsensus = null;
     }
 
+    /**
+     * Creates new transaction record builder for a child transaction.
+     *
+     * @param consensusNow    the consensus timestamp for the transaction
+     * @param parentConsensus the consensus timestamp of the parent transaction
+     */
     public SingleTransactionRecordBuilderImpl(
-            @NonNull final Instant consensusNow, @NonNull final Instant parentConsensusTimestamp) {
+            @NonNull final Instant consensusNow, @NonNull final Instant parentConsensus) {
         this.consensusNow = requireNonNull(consensusNow, "consensusNow must not be null");
-        this.parentConsensusTimestamp =
-                requireNonNull(parentConsensusTimestamp, "parentConsensusTimestamp must not be null");
+        this.parentConsensus = requireNonNull(parentConsensus, "parentConsensusTimestamp must not be null");
```

Comment:
You can also change the error description from "parentConsensusTimestamp must not be null" to "parentConsensus must not be null"

Reply:
Yes, we should do this in a second PR